<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Week 4 Genetic Diversity | EVE 109: Molecular Ecology</title>
  <meta name="description" content="Tutorials and datasets for discussion and homeowork excesises for EVE 109: Molecular Ecology" />
  <meta name="generator" content="bookdown 0.22 and GitBook 2.6.7" />

  <meta property="og:title" content="Week 4 Genetic Diversity | EVE 109: Molecular Ecology" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="Tutorials and datasets for discussion and homeowork excesises for EVE 109: Molecular Ecology" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Week 4 Genetic Diversity | EVE 109: Molecular Ecology" />
  
  <meta name="twitter:description" content="Tutorials and datasets for discussion and homeowork excesises for EVE 109: Molecular Ecology" />
  




  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="genetic-diversity.html"/>

<script src="libs/header-attrs-2.8/header-attrs.js"></script>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">EVE109 FQ21</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#installing-r-and-rstudio"><i class="fa fa-check"></i><b>1.1</b> Installing R and RStudio</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#tour-of-rstudio"><i class="fa fa-check"></i><b>1.2</b> Tour of RStudio</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#installing-packages"><i class="fa fa-check"></i><b>1.3</b> Installing packages</a></li>
<li class="chapter" data-level="1.4" data-path="index.html"><a href="index.html#basic-functions-in-r"><i class="fa fa-check"></i><b>1.4</b> Basic functions in R</a>
<ul>
<li class="chapter" data-level="1.4.1" data-path="index.html"><a href="index.html#exercise-write-a-script-to-find-out-what-fraction-of-your-life-youve-been-at-uc-davis"><i class="fa fa-check"></i><b>1.4.1</b> <em>Exercise</em>: Write a script to find out what fraction of your life you’ve been at UC Davis</a></li>
</ul></li>
<li class="chapter" data-level="1.5" data-path="index.html"><a href="index.html#classes"><i class="fa fa-check"></i><b>1.5</b> Classes</a></li>
<li class="chapter" data-level="1.6" data-path="index.html"><a href="index.html#vectors-and-dataframes"><i class="fa fa-check"></i><b>1.6</b> Vectors and dataframes</a></li>
<li class="chapter" data-level="1.7" data-path="index.html"><a href="index.html#subsetting"><i class="fa fa-check"></i><b>1.7</b> Subsetting</a></li>
<li class="chapter" data-level="1.8" data-path="index.html"><a href="index.html#plotting"><i class="fa fa-check"></i><b>1.8</b> Plotting</a>
<ul>
<li class="chapter" data-level="1.8.1" data-path="index.html"><a href="index.html#exercise-describe-this-plot-in-plain-english.-what-conclusions-can-we-draw"><i class="fa fa-check"></i><b>1.8.1</b> <em>Exercise</em>: Describe this plot in plain english. What conclusions can we draw?</a></li>
</ul></li>
<li class="chapter" data-level="1.9" data-path="index.html"><a href="index.html#homework"><i class="fa fa-check"></i><b>1.9</b> Homework</a>
<ul>
<li class="chapter" data-level="1.9.1" data-path="index.html"><a href="index.html#homework-1-write-a-script-that-does-the-following"><i class="fa fa-check"></i><b>1.9.1</b> <em>Homework 1: Write a script that does the following:</em></a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="2" data-path="molecular-markers.html"><a href="molecular-markers.html"><i class="fa fa-check"></i><b>2</b> Molecular Markers</a>
<ul>
<li class="chapter" data-level="2.1" data-path="molecular-markers.html"><a href="molecular-markers.html#paths"><i class="fa fa-check"></i><b>2.1</b> Paths</a></li>
<li class="chapter" data-level="2.2" data-path="molecular-markers.html"><a href="molecular-markers.html#reading-in-data-from-spreadsheets"><i class="fa fa-check"></i><b>2.2</b> Reading in data from spreadsheets</a></li>
<li class="chapter" data-level="2.3" data-path="molecular-markers.html"><a href="molecular-markers.html#simple-capture-recapture-example"><i class="fa fa-check"></i><b>2.3</b> Simple capture recapture example</a></li>
<li class="chapter" data-level="2.4" data-path="molecular-markers.html"><a href="molecular-markers.html#homework-1"><i class="fa fa-check"></i><b>2.4</b> Homework</a>
<ul>
<li class="chapter" data-level="2.4.1" data-path="molecular-markers.html"><a href="molecular-markers.html#homework-2-write-a-script-that-does-the-following"><i class="fa fa-check"></i><b>2.4.1</b> <em>Homework 2: Write a script that does the following:</em></a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="genetic-diversity.html"><a href="genetic-diversity.html"><i class="fa fa-check"></i><b>3</b> Genetic Diversity</a>
<ul>
<li class="chapter" data-level="3.1" data-path="genetic-diversity.html"><a href="genetic-diversity.html#manipulating-dataframes"><i class="fa fa-check"></i><b>3.1</b> Manipulating dataframes</a></li>
<li class="chapter" data-level="3.2" data-path="genetic-diversity.html"><a href="genetic-diversity.html#using-the-adegenet-package-to-calulate-heterozygosity"><i class="fa fa-check"></i><b>3.2</b> Using the <strong>adegenet</strong> package to calulate heterozygosity</a>
<ul>
<li class="chapter" data-level="3.2.1" data-path="genetic-diversity.html"><a href="genetic-diversity.html#plotting-heterozygosity"><i class="fa fa-check"></i><b>3.2.1</b> Plotting heterozygosity</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="genetic-diversity.html"><a href="genetic-diversity.html#homework-2"><i class="fa fa-check"></i><b>3.3</b> Homework</a>
<ul>
<li class="chapter" data-level="3.3.1" data-path="genetic-diversity.html"><a href="genetic-diversity.html#homework-3-write-a-script-that-does-the-following"><i class="fa fa-check"></i><b>3.3.1</b> <em>Homework 3: Write a script that does the following:</em></a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="genetic-diversity-1.html"><a href="genetic-diversity-1.html"><i class="fa fa-check"></i><b>4</b> Genetic Diversity</a>
<ul>
<li class="chapter" data-level="4.1" data-path="genetic-diversity-1.html"><a href="genetic-diversity-1.html#the-aggregate-and-apply-functions"><i class="fa fa-check"></i><b>4.1</b> The <code>aggregate</code> and <code>apply</code> functions</a></li>
<li class="chapter" data-level="4.2" data-path="genetic-diversity-1.html"><a href="genetic-diversity-1.html#running-a-clustering-analysis-and-deciding-on-a-k-value"><i class="fa fa-check"></i><b>4.2</b> Running a clustering analysis and deciding on a K value</a></li>
<li class="chapter" data-level="4.3" data-path="genetic-diversity-1.html"><a href="genetic-diversity-1.html#the-q-matrix"><i class="fa fa-check"></i><b>4.3</b> The Q matrix</a></li>
<li class="chapter" data-level="4.4" data-path="genetic-diversity-1.html"><a href="genetic-diversity-1.html#plotting-1"><i class="fa fa-check"></i><b>4.4</b> Plotting</a></li>
<li class="chapter" data-level="4.5" data-path="genetic-diversity-1.html"><a href="genetic-diversity-1.html#homework-3"><i class="fa fa-check"></i><b>4.5</b> Homework</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">EVE 109: Molecular Ecology</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="genetic-diversity-1" class="section level1" number="4">
<h1><span class="header-section-number">Week 4</span> Genetic Diversity</h1>
<div id="the-aggregate-and-apply-functions" class="section level2" number="4.1">
<h2><span class="header-section-number">4.1</span> The <code>aggregate</code> and <code>apply</code> functions</h2>
<p>Today we’ll two useful functions that you can use for dataframes, <code>apply</code> and <code>aggregate</code>. Let’s use the <code>palmerpenguins</code> dataset again:</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="genetic-diversity-1.html#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(palmerpenguins)</span>
<span id="cb1-2"><a href="genetic-diversity-1.html#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(penguins)</span></code></pre></div>
<pre><code>## # A tibble: 6 x 8
##   species island bill_length_mm bill_depth_mm flipper_length_… body_mass_g sex  
##   &lt;fct&gt;   &lt;fct&gt;           &lt;dbl&gt;         &lt;dbl&gt;            &lt;int&gt;       &lt;int&gt; &lt;fct&gt;
## 1 Adelie  Torge…           39.1          18.7              181        3750 male 
## 2 Adelie  Torge…           39.5          17.4              186        3800 fema…
## 3 Adelie  Torge…           40.3          18                195        3250 fema…
## 4 Adelie  Torge…           NA            NA                 NA          NA &lt;NA&gt; 
## 5 Adelie  Torge…           36.7          19.3              193        3450 fema…
## 6 Adelie  Torge…           39.3          20.6              190        3650 male 
## # … with 1 more variable: year &lt;int&gt;</code></pre>
<p> </p>
<p>One thing we might want to do is find out which individual has the highest value for each column. We can use the <code>apply</code> function to do the same thing to each column:</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="genetic-diversity-1.html#cb3-1" aria-hidden="true" tabindex="-1"></a>?apply</span>
<span id="cb3-2"><a href="genetic-diversity-1.html#cb3-2" aria-hidden="true" tabindex="-1"></a><span class="fu">apply</span>(penguins[,<span class="dv">3</span><span class="sc">:</span><span class="dv">6</span>],<span class="at">MARGIN=</span><span class="dv">2</span>,which.max)</span></code></pre></div>
<pre><code>##    bill_length_mm     bill_depth_mm flipper_length_mm       body_mass_g 
##               186                20               216               170</code></pre>
<p>Notice that we only used columns 3-6 because these are numeric (rather than categorical). The second argument, <em>MARGIN</em>, is asking whether we should be doing the function for each row (MARGIN=1), or each column (MARGIN=2). The final argument is the function we want to use. <code>which.max</code> asks for the index (in this case, which row) has the maximum value.</p>
<p> </p>
<p>Another useful function is <code>aggregate</code>. Sometimes you might want to summarize groups within a dataframe. <code>aggregate</code> does this easily. For example, what if we want to average values for each species:</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="genetic-diversity-1.html#cb5-1" aria-hidden="true" tabindex="-1"></a>?aggregate</span>
<span id="cb5-2"><a href="genetic-diversity-1.html#cb5-2" aria-hidden="true" tabindex="-1"></a><span class="fu">aggregate</span>(penguins[,<span class="dv">3</span><span class="sc">:</span><span class="dv">6</span>],<span class="at">by=</span><span class="fu">list</span>(penguins<span class="sc">$</span>species),mean)</span></code></pre></div>
<pre><code>##     Group.1 bill_length_mm bill_depth_mm flipper_length_mm body_mass_g
## 1    Adelie             NA            NA                NA          NA
## 2 Chinstrap       48.83382      18.42059          195.8235    3733.088
## 3    Gentoo             NA            NA                NA          NA</code></pre>
<p>This gives us the mean for each set of rows that share a value in the “species” column. The second argument, <em>by</em>, tells how to group. The third argument is the function to use.</p>
<p> </p>
<p>These commands will come in handy today!</p>
<p>Let’s take just the quantitative data for a moment:</p>
<p> 
 
 </p>
</div>
<div id="running-a-clustering-analysis-and-deciding-on-a-k-value" class="section level2" number="4.2">
<h2><span class="header-section-number">4.2</span> Running a clustering analysis and deciding on a K value</h2>
<p>The example file for use in class today can be downloaded <a href="data/week4_example.geno">here</a></p>
<p>A common way to analyze population structure is using hierarchical clustering. Probably the most heavily used program to do this is called STRUCTURE. However, we’ll stick to similar tools you can use in R. Today we’ll use <code>LEA</code> for our clustering analysis. LEA cannot be downloaded from the normal R database, so use the following code:</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="genetic-diversity-1.html#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">&quot;BiocManager&quot;</span>)</span>
<span id="cb7-2"><a href="genetic-diversity-1.html#cb7-2" aria-hidden="true" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">&quot;LEA&quot;</span>)</span></code></pre></div>
<p>Then load the library</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="genetic-diversity-1.html#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(LEA)</span></code></pre></div>
<p> </p>
<p>For the analysis today, we’re going to use SNP data. Open the file you downloaded called <strong>week4_example.geno</strong> using a text editor. This file is in <em>geno</em> format. That means there is one line per individual and genotypes are coded as 0,1,or 2. 0 and 2 are homozygotes and 1 is a heterozygote. 9 means missing data. Notice that this only works when you are looking at SNPs with just 2 alleles (biallelic).</p>
<p> </p>
<p>We’ll use the package LEA to read this file in and look at population structure. Make sure you have the right path!!</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="genetic-diversity-1.html#cb9-1" aria-hidden="true" tabindex="-1"></a>project <span class="ot">&lt;-</span> <span class="fu">snmf</span>(<span class="st">&quot;data/week4_example.geno&quot;</span>,</span>
<span id="cb9-2"><a href="genetic-diversity-1.html#cb9-2" aria-hidden="true" tabindex="-1"></a>               <span class="at">K =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>, </span>
<span id="cb9-3"><a href="genetic-diversity-1.html#cb9-3" aria-hidden="true" tabindex="-1"></a>               <span class="at">entropy =</span> <span class="cn">TRUE</span>, </span>
<span id="cb9-4"><a href="genetic-diversity-1.html#cb9-4" aria-hidden="true" tabindex="-1"></a>               <span class="at">repetitions =</span> <span class="dv">10</span>,</span>
<span id="cb9-5"><a href="genetic-diversity-1.html#cb9-5" aria-hidden="true" tabindex="-1"></a>               <span class="at">project =</span> <span class="st">&quot;new&quot;</span>)</span></code></pre></div>
<p>Remember what K values are? That means ‘how many populations should we try to split this data in to?’ Here we run 10 <em>repetitions</em> per K value. That’s because each run is slightly different so you want to have confidence in your answer. You’ll see what entropy means in a second. You’ll get a bunch of output in your console when you run this command. It will also create a new folder on your computer called “example.snmf.” It contains a bunch of different files for each run of the clustering program. We will be able to use these files to analyze and visualize our results.</p>
<p> </p>
<p>Now we’ll chose the “best” value of K. Know that there are several ways to do this and they don’t always agree, so this should be interpreted with caution. Here we will use cross-entropy loss. This basically asks how consistently our model is able to categorize the samples. The lower the cross-entropy the better the model is doing. Let’s plot all the values of K:</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="genetic-diversity-1.html#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(project,<span class="at">col=</span><span class="st">&quot;blue&quot;</span>,<span class="at">pch=</span><span class="dv">19</span>)</span></code></pre></div>
<p><img src="Website_files/figure-html/unnamed-chunk-7-1.png" width="672" /></p>
<p> </p>
<p>When I ran this, <strong>K=4</strong> had the lowest cross entropy. This plot might look slightly different each time you run it.</p>
<p> 
 
 </p>
</div>
<div id="the-q-matrix" class="section level2" number="4.3">
<h2><span class="header-section-number">4.3</span> The Q matrix</h2>
<p>Now that we’ve decided which value of K to look at, we can look at the ancestry estimates for each individual.</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="genetic-diversity-1.html#cb11-1" aria-hidden="true" tabindex="-1"></a>qmatrix <span class="ot">&lt;-</span> <span class="fu">Q</span>(project,<span class="at">K=</span><span class="dv">4</span>,<span class="at">run=</span><span class="dv">1</span>)</span></code></pre></div>
<p> </p>
<p>This command pulls out the ancestry matrix (called the <em>Q matrix</em>) for the first run with K=3. Let’s look at the Q matrix.</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="genetic-diversity-1.html#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(qmatrix)</span></code></pre></div>
<pre><code>##               V1          V2          V3          V4
## [1,] 0.000099991 0.768583000 0.149645000 0.081672100
## [2,] 0.000099982 0.964339000 0.000099982 0.035461300
## [3,] 0.000099982 0.545125000 0.454675000 0.000099982
## [4,] 0.211788000 0.680727000 0.107385000 0.000099991
## [5,] 0.000099982 0.000099982 0.587792000 0.412008000
## [6,] 0.939237000 0.003872600 0.056790100 0.000099991</code></pre>
<p> </p>
<p>Here, each row is an individual and each column is one of the inferred clusters. First, lets give the individuals “IDs” as rownames. You’ll see that this is important later. Next, one thing we could do is look at which cluster has the highest ancestry for each individual. The <code>which.max</code> function just gives you the index (which column in this case) of the highest value.</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="genetic-diversity-1.html#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(qmatrix) <span class="ot">&lt;-</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(qmatrix)</span>
<span id="cb14-2"><a href="genetic-diversity-1.html#cb14-2" aria-hidden="true" tabindex="-1"></a>maxCluster <span class="ot">&lt;-</span> <span class="fu">apply</span>(qmatrix,<span class="dv">1</span>,which.max)</span>
<span id="cb14-3"><a href="genetic-diversity-1.html#cb14-3" aria-hidden="true" tabindex="-1"></a>maxCluster</span></code></pre></div>
<pre><code>##  1  2  3  4  5  6  7  8  9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 
##  2  2  2  2  3  1  3  2  1  3  3  2  4  3  4  4  2  4  1  4  4  1  4  2  3  4 
## 27 28 29 30 31 32 33 34 35 36 37 38 39 40 41 42 43 44 45 46 47 48 49 50 
##  2  1  2  1  3  3  3  1  2  4  1  1  3  1  1  2  4  3  4  2  4  1  4  4</code></pre>
<p> </p>
<p>If we pretend <code>maxCluster</code> gives refers to the population from which the individual came, we can use the <code>aggregate</code> function to look at average cluster membership across each population:</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="genetic-diversity-1.html#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="fu">aggregate</span>(qmatrix,<span class="at">by=</span><span class="fu">list</span>(maxCluster),mean)</span></code></pre></div>
<pre><code>##   Group.1         V1         V2         V3         V4
## 1       1 0.80271658 0.07627143 0.03502381 0.08598811
## 2       2 0.03968297 0.84556923 0.09996497 0.01478282
## 3       3 0.04753737 0.01866167 0.84999500 0.08380596
## 4       4 0.13964038 0.01670342 0.12993916 0.71371700</code></pre>
<p>NOTE: Although we are using <code>maxCluster</code> as an indicator of population here, usually we would use some sort of <strong>a priori</strong> information we had, such as geographic location or maybe morphological species, depending on our question. This distinction will be important for your homework!</p>
<p> 
 
 </p>
</div>
<div id="plotting-1" class="section level2" number="4.4">
<h2><span class="header-section-number">4.4</span> Plotting</h2>
<p>Now lets plot the ancestry results. Basically, we just want to make a barplot of the qmatrix. You can do that with the <code>barchart</code> function in the LEA package:</p>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="genetic-diversity-1.html#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="fu">barchart</span>(project, <span class="at">K=</span><span class="dv">4</span>, <span class="at">run=</span><span class="dv">1</span>,</span>
<span id="cb18-2"><a href="genetic-diversity-1.html#cb18-2" aria-hidden="true" tabindex="-1"></a>         <span class="at">border=</span><span class="cn">NA</span>, <span class="at">space=</span><span class="dv">0</span>)</span></code></pre></div>
<p><img src="Website_files/figure-html/unnamed-chunk-12-1.png" width="672" /></p>
<pre><code>## $order
##  [1]  5  7 10 11 14 25 31 32 33 39 44  1  2  3  4  8 12 17 24 27 29 35 42 46 13
## [26] 15 16 18 20 21 23 26 36 43 45 47 49 50  6  9 19 22 28 30 34 37 38 40 41 48</code></pre>
<p>We can also try to label each bar with information. For example, lets try to label with the population assignment that we inferred, <code>maxCluster</code>:</p>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb20-1"><a href="genetic-diversity-1.html#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="fu">barchart</span>(project, <span class="at">K=</span><span class="dv">4</span>, <span class="at">run=</span><span class="dv">1</span>,</span>
<span id="cb20-2"><a href="genetic-diversity-1.html#cb20-2" aria-hidden="true" tabindex="-1"></a>         <span class="at">border=</span><span class="cn">NA</span>, <span class="at">space=</span><span class="dv">0</span>) <span class="ot">-&gt;</span> bp</span>
<span id="cb20-3"><a href="genetic-diversity-1.html#cb20-3" aria-hidden="true" tabindex="-1"></a><span class="fu">axis</span>(<span class="dv">1</span>,<span class="at">at=</span><span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(bp<span class="sc">$</span>order),<span class="at">label=</span>maxCluster[bp<span class="sc">$</span>order],<span class="at">las=</span><span class="dv">2</span>,<span class="at">cex.axis=</span><span class="fl">0.4</span>)</span></code></pre></div>
<p><img src="Website_files/figure-html/unnamed-chunk-13-1.png" width="672" /></p>
<p> 
The grayscale is boring!! R has many many options for color packages and you could even make one of your own! For example, <a href="https://github.com/johannesbjork/LaCroixColoR">here is a fun one inspired by favorite La Croix flavors!</a></p>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="genetic-diversity-1.html#cb21-1" aria-hidden="true" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">&quot;devtools&quot;</span>)</span>
<span id="cb21-2"><a href="genetic-diversity-1.html#cb21-2" aria-hidden="true" tabindex="-1"></a>devtools<span class="sc">::</span><span class="fu">install_github</span>(<span class="st">&quot;johannesbjork/LaCroixColoR&quot;</span>)</span></code></pre></div>
<p>And now use it in our bar chart:</p>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="genetic-diversity-1.html#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(LaCroixColoR)</span>
<span id="cb22-2"><a href="genetic-diversity-1.html#cb22-2" aria-hidden="true" tabindex="-1"></a>my.cols<span class="ot">=</span><span class="fu">lacroix_palette</span>(<span class="st">&quot;PassionFruit&quot;</span>,<span class="at">n=</span><span class="dv">4</span>,<span class="at">type=</span><span class="st">&quot;discrete&quot;</span>)</span>
<span id="cb22-3"><a href="genetic-diversity-1.html#cb22-3" aria-hidden="true" tabindex="-1"></a><span class="fu">barchart</span>(project, <span class="at">K=</span><span class="dv">4</span>, <span class="at">run=</span><span class="dv">1</span>,</span>
<span id="cb22-4"><a href="genetic-diversity-1.html#cb22-4" aria-hidden="true" tabindex="-1"></a>         <span class="at">border=</span><span class="cn">NA</span>, <span class="at">space=</span><span class="dv">0</span>,<span class="at">col=</span>my.cols) <span class="ot">-&gt;</span> bp</span>
<span id="cb22-5"><a href="genetic-diversity-1.html#cb22-5" aria-hidden="true" tabindex="-1"></a><span class="fu">axis</span>(<span class="dv">1</span>,<span class="at">at=</span><span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(bp<span class="sc">$</span>order),<span class="at">label=</span>maxCluster[bp<span class="sc">$</span>order],<span class="at">las=</span><span class="dv">2</span>,<span class="at">cex.axis=</span><span class="fl">0.4</span>)</span></code></pre></div>
<p><img src="Website_files/figure-html/unnamed-chunk-15-1.png" width="672" /></p>
</div>
<div id="homework-3" class="section level2" number="4.5">
<h2><span class="header-section-number">4.5</span> Homework</h2>
<p>Now that we know how to estimate observed and expected heterozygosity, we’ll look at changes over time. Start a new script for your homework</p>
<p>###<em>Homework 3: Write a script that does the following:</em>
1. Read in “genotypes.csv” and subset the dataframe to Mt Buller samples only. How many samples do you have from each year?
2. Calculate mean observed heterozygosity for each year. Put these into a single dataframe.
3. Calculate mean expected heterozygosity for each year. Add these to the dataframe from question 2.
4. Plot mean expected heterozygosity over time.
5. Add a vertical lines to your plot to show when introductions took place (hint - use the <code>abline</code> command)</p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="genetic-diversity.html" class="navigation navigation-prev navigation-unique" aria-label="Previous page"><i class="fa fa-angle-left"></i></a>

    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["Website.pdf", "Website.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
