<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Week 2 Molecular Markers | EVE 109: Molecular Ecology</title>
  <meta name="description" content="Tutorials and datasets for discussion and homeowork excesises for EVE 109: Molecular Ecology" />
  <meta name="generator" content="bookdown 0.24 and GitBook 2.6.7" />

  <meta property="og:title" content="Week 2 Molecular Markers | EVE 109: Molecular Ecology" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="Tutorials and datasets for discussion and homeowork excesises for EVE 109: Molecular Ecology" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Week 2 Molecular Markers | EVE 109: Molecular Ecology" />
  
  <meta name="twitter:description" content="Tutorials and datasets for discussion and homeowork excesises for EVE 109: Molecular Ecology" />
  




  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="index.html"/>
<link rel="next" href="genetic-diversity.html"/>
<script src="libs/header-attrs-2.11/header-attrs.js"></script>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">EVE109 FQ21</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#installing-r-and-rstudio"><i class="fa fa-check"></i><b>1.1</b> Installing R and RStudio</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#tour-of-rstudio"><i class="fa fa-check"></i><b>1.2</b> Tour of RStudio</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#installing-packages"><i class="fa fa-check"></i><b>1.3</b> Installing packages</a></li>
<li class="chapter" data-level="1.4" data-path="index.html"><a href="index.html#basic-functions-in-r"><i class="fa fa-check"></i><b>1.4</b> Basic functions in R</a>
<ul>
<li class="chapter" data-level="1.4.1" data-path="index.html"><a href="index.html#exercise-write-a-script-to-find-out-what-fraction-of-your-life-youve-been-at-uc-davis"><i class="fa fa-check"></i><b>1.4.1</b> <em>Exercise</em>: Write a script to find out what fraction of your life you’ve been at UC Davis</a></li>
</ul></li>
<li class="chapter" data-level="1.5" data-path="index.html"><a href="index.html#classes"><i class="fa fa-check"></i><b>1.5</b> Classes</a></li>
<li class="chapter" data-level="1.6" data-path="index.html"><a href="index.html#vectors-and-dataframes"><i class="fa fa-check"></i><b>1.6</b> Vectors and dataframes</a></li>
<li class="chapter" data-level="1.7" data-path="index.html"><a href="index.html#subsetting"><i class="fa fa-check"></i><b>1.7</b> Subsetting</a></li>
<li class="chapter" data-level="1.8" data-path="index.html"><a href="index.html#plotting"><i class="fa fa-check"></i><b>1.8</b> Plotting</a>
<ul>
<li class="chapter" data-level="1.8.1" data-path="index.html"><a href="index.html#exercise-describe-this-plot-in-plain-english.-what-conclusions-can-we-draw"><i class="fa fa-check"></i><b>1.8.1</b> <em>Exercise</em>: Describe this plot in plain english. What conclusions can we draw?</a></li>
</ul></li>
<li class="chapter" data-level="1.9" data-path="index.html"><a href="index.html#homework"><i class="fa fa-check"></i><b>1.9</b> Homework</a>
<ul>
<li class="chapter" data-level="1.9.1" data-path="index.html"><a href="index.html#homework-1-write-a-script-that-does-the-following"><i class="fa fa-check"></i><b>1.9.1</b> <em>Homework 1: Write a script that does the following:</em></a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="2" data-path="molecular-markers.html"><a href="molecular-markers.html"><i class="fa fa-check"></i><b>2</b> Molecular Markers</a>
<ul>
<li class="chapter" data-level="2.1" data-path="molecular-markers.html"><a href="molecular-markers.html#paths"><i class="fa fa-check"></i><b>2.1</b> Paths</a></li>
<li class="chapter" data-level="2.2" data-path="molecular-markers.html"><a href="molecular-markers.html#reading-in-data-from-spreadsheets"><i class="fa fa-check"></i><b>2.2</b> Reading in data from spreadsheets</a></li>
<li class="chapter" data-level="2.3" data-path="molecular-markers.html"><a href="molecular-markers.html#simple-capture-recapture-example"><i class="fa fa-check"></i><b>2.3</b> Simple capture recapture example</a></li>
<li class="chapter" data-level="2.4" data-path="molecular-markers.html"><a href="molecular-markers.html#homework-1"><i class="fa fa-check"></i><b>2.4</b> Homework</a>
<ul>
<li class="chapter" data-level="2.4.1" data-path="molecular-markers.html"><a href="molecular-markers.html#homework-2-write-a-script-that-does-the-following"><i class="fa fa-check"></i><b>2.4.1</b> <em>Homework 2: Write a script that does the following:</em></a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="genetic-diversity.html"><a href="genetic-diversity.html"><i class="fa fa-check"></i><b>3</b> Genetic Diversity</a>
<ul>
<li class="chapter" data-level="3.1" data-path="genetic-diversity.html"><a href="genetic-diversity.html#manipulating-dataframes"><i class="fa fa-check"></i><b>3.1</b> Manipulating dataframes</a></li>
<li class="chapter" data-level="3.2" data-path="genetic-diversity.html"><a href="genetic-diversity.html#using-the-adegenet-package-to-calulate-heterozygosity"><i class="fa fa-check"></i><b>3.2</b> Using the <strong>adegenet</strong> package to calulate heterozygosity</a>
<ul>
<li class="chapter" data-level="3.2.1" data-path="genetic-diversity.html"><a href="genetic-diversity.html#plotting-heterozygosity"><i class="fa fa-check"></i><b>3.2.1</b> Plotting heterozygosity</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="genetic-diversity.html"><a href="genetic-diversity.html#homework-2"><i class="fa fa-check"></i><b>3.3</b> Homework</a>
<ul>
<li class="chapter" data-level="3.3.1" data-path="genetic-diversity.html"><a href="genetic-diversity.html#homework-3-write-a-script-that-does-the-following"><i class="fa fa-check"></i><b>3.3.1</b> <em>Homework 3: Write a script that does the following:</em></a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="population-structure.html"><a href="population-structure.html"><i class="fa fa-check"></i><b>4</b> Population Structure</a>
<ul>
<li class="chapter" data-level="4.1" data-path="population-structure.html"><a href="population-structure.html#the-aggregate-and-apply-functions"><i class="fa fa-check"></i><b>4.1</b> The <code>aggregate</code> and <code>apply</code> functions</a></li>
<li class="chapter" data-level="4.2" data-path="population-structure.html"><a href="population-structure.html#running-a-clustering-analysis-and-deciding-on-a-k-value"><i class="fa fa-check"></i><b>4.2</b> Running a clustering analysis and deciding on a K value</a></li>
<li class="chapter" data-level="4.3" data-path="population-structure.html"><a href="population-structure.html#the-q-matrix"><i class="fa fa-check"></i><b>4.3</b> The Q matrix</a></li>
<li class="chapter" data-level="4.4" data-path="population-structure.html"><a href="population-structure.html#plotting-1"><i class="fa fa-check"></i><b>4.4</b> Plotting</a></li>
<li class="chapter" data-level="4.5" data-path="population-structure.html"><a href="population-structure.html#homework-3"><i class="fa fa-check"></i><b>4.5</b> Homework</a>
<ul>
<li class="chapter" data-level="4.5.1" data-path="population-structure.html"><a href="population-structure.html#homework-4-write-a-script-that-does-the-following"><i class="fa fa-check"></i><b>4.5.1</b> <em>Homework 4: Write a script that does the following:</em></a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="haplotype-networks.html"><a href="haplotype-networks.html"><i class="fa fa-check"></i><b>5</b> Haplotype networks</a>
<ul>
<li class="chapter" data-level="5.1" data-path="haplotype-networks.html"><a href="haplotype-networks.html#indexing"><i class="fa fa-check"></i><b>5.1</b> Indexing</a></li>
<li class="chapter" data-level="5.2" data-path="haplotype-networks.html"><a href="haplotype-networks.html#using-table-to-count-across-multiple-columns"><i class="fa fa-check"></i><b>5.2</b> Using <code>table</code> to count across multiple columns</a></li>
<li class="chapter" data-level="5.3" data-path="haplotype-networks.html"><a href="haplotype-networks.html#creating-a-haplotype-network"><i class="fa fa-check"></i><b>5.3</b> Creating a haplotype network</a></li>
<li class="chapter" data-level="5.4" data-path="haplotype-networks.html"><a href="haplotype-networks.html#homework-4"><i class="fa fa-check"></i><b>5.4</b> Homework</a>
<ul>
<li class="chapter" data-level="5.4.1" data-path="haplotype-networks.html"><a href="haplotype-networks.html#homework-5-write-a-script-that-does-the-following"><i class="fa fa-check"></i><b>5.4.1</b> <em>Homework 5: Write a script that does the following:</em></a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6" data-path="project-time.html"><a href="project-time.html"><i class="fa fa-check"></i><b>6</b> Project Time!</a></li>
<li class="chapter" data-level="7" data-path="speciation-and-hybridization.html"><a href="speciation-and-hybridization.html"><i class="fa fa-check"></i><b>7</b> Speciation and Hybridization</a>
<ul>
<li class="chapter" data-level="7.1" data-path="speciation-and-hybridization.html"><a href="speciation-and-hybridization.html#match"><i class="fa fa-check"></i><b>7.1</b> Match</a></li>
<li class="chapter" data-level="7.2" data-path="speciation-and-hybridization.html"><a href="speciation-and-hybridization.html#dealing-with-missing-data"><i class="fa fa-check"></i><b>7.2</b> Dealing with missing data</a></li>
<li class="chapter" data-level="7.3" data-path="speciation-and-hybridization.html"><a href="speciation-and-hybridization.html#writing-your-own-functions"><i class="fa fa-check"></i><b>7.3</b> Writing your own functions</a></li>
<li class="chapter" data-level="7.4" data-path="speciation-and-hybridization.html"><a href="speciation-and-hybridization.html#hybrid-allele-frequencies"><i class="fa fa-check"></i><b>7.4</b> Hybrid allele frequencies</a></li>
<li class="chapter" data-level="7.5" data-path="speciation-and-hybridization.html"><a href="speciation-and-hybridization.html#making-a-pca"><i class="fa fa-check"></i><b>7.5</b> Making a PCA</a></li>
<li class="chapter" data-level="7.6" data-path="speciation-and-hybridization.html"><a href="speciation-and-hybridization.html#homework-5"><i class="fa fa-check"></i><b>7.6</b> Homework</a>
<ul>
<li class="chapter" data-level="7.6.1" data-path="speciation-and-hybridization.html"><a href="speciation-and-hybridization.html#homework-5-write-a-script-that-does-the-following-1"><i class="fa fa-check"></i><b>7.6.1</b> <em>Homework 5: Write a script that does the following:</em></a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="8" data-path="veterans-day.html"><a href="veterans-day.html"><i class="fa fa-check"></i><b>8</b> Veteran’s Day</a></li>
<li class="chapter" data-level="9" data-path="sex-biased-gene-flow.html"><a href="sex-biased-gene-flow.html"><i class="fa fa-check"></i><b>9</b> Sex-biased gene flow</a>
<ul>
<li class="chapter" data-level="9.1" data-path="sex-biased-gene-flow.html"><a href="sex-biased-gene-flow.html#multiple-plots"><i class="fa fa-check"></i><b>9.1</b> Multiple plots</a></li>
<li class="chapter" data-level="9.2" data-path="sex-biased-gene-flow.html"><a href="sex-biased-gene-flow.html#intro-to-this-weeks-data"><i class="fa fa-check"></i><b>9.2</b> Intro to this week’s data</a></li>
<li class="chapter" data-level="9.3" data-path="sex-biased-gene-flow.html"><a href="sex-biased-gene-flow.html#homework-6"><i class="fa fa-check"></i><b>9.3</b> Homework</a>
<ul>
<li class="chapter" data-level="9.3.1" data-path="sex-biased-gene-flow.html"><a href="sex-biased-gene-flow.html#homework-8-write-a-script-that-does-the-following"><i class="fa fa-check"></i><b>9.3.1</b> <em>Homework 8: Write a script that does the following:</em></a></li>
</ul></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">EVE 109: Molecular Ecology</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="molecular-markers" class="section level1" number="2">
<h1><span class="header-section-number">Week 2</span> Molecular Markers</h1>
<div id="paths" class="section level2" number="2.1">
<h2><span class="header-section-number">2.1</span> Paths</h2>
<p>For this course, I will upload the datasets you need to this website and you will download them to your own computer. You will then need to figure out how tell R where to find them. You can find out what directory you are currently in using the <code>getwd()</code> command:</p>
<div class="sourceCode" id="cb42"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb42-1"><a href="molecular-markers.html#cb42-1" aria-hidden="true" tabindex="-1"></a><span class="fu">getwd</span>()</span></code></pre></div>
<pre><code>## [1] &quot;/Users/rachaelbay/Documents/Courses/Molecular Ecology/2021/Website&quot;</code></pre>
<p>Directories are hierarchical: the represent folders within folders just like you view them when you navigate with your mouse. You can also set which directory R will use to look for files. This is called your ‘working directory’ and you can set it using <code>setwd()</code>. For example, if I create a folder called "EVE109 inside my documents folder where I want to store all my materials for this class, I can then tell R I want it to look in that directory for files:</p>
<div class="sourceCode" id="cb44"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb44-1"><a href="molecular-markers.html#cb44-1" aria-hidden="true" tabindex="-1"></a><span class="fu">setwd</span>(<span class="st">&quot;~/Documents/EVE109&quot;</span>)</span></code></pre></div>
<p>If you get an error that your file cannot be found, a good first step is to check your working directory.</p>
<p> 
 
 </p>
</div>
<div id="reading-in-data-from-spreadsheets" class="section level2" number="2.2">
<h2><span class="header-section-number">2.2</span> Reading in data from spreadsheets</h2>
<p>Today we will be working with data from the paper we read in class. You can download the dataset <a href="data/wombats.csv">here</a></p>
<p>A lot of data we use is stored in spreadsheets. However, when we store them in a normal Excel format (.xlsx) they come with a lot of extra formatting that R has trouble reading. One way to avoid this is to store them in a comma separated value format (.csv). Open “wombats.csv” in a text editor and look at it. It’s not easy for us to read, but you can see that it is a very simple format where each entry is separated by a comma. You were able to open it in a spreadsheet program and R can read it easily. If you are making your own data, you can save spreadsheets as .csv files from most spreadsheet programs.</p>
<p>Now we want to read that data into RStudio. We can do that using the command <code>read.csv</code>. Remember do assign the resulting data frame to an object:</p>
<div class="sourceCode" id="cb45"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb45-1"><a href="molecular-markers.html#cb45-1" aria-hidden="true" tabindex="-1"></a>data <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">&quot;data/wombats.csv&quot;</span>) <span class="co"># Read in a file</span></span></code></pre></div>
<p>Remember, we can look at the top of the dataframe using <code>head</code>:</p>
<div class="sourceCode" id="cb46"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb46-1"><a href="molecular-markers.html#cb46-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(data)</span></code></pre></div>
<pre><code>##   No.captures total males females
## 1           1    25    18       7
## 2           2    11     7       4
## 3           3    10     7       3
## 4           4    10     6       4
## 5           5     9     6       3
## 6           6    11     6       5</code></pre>
<p>This is data from the paper we read this week by Banks et al. (2003) using genetic analysis to monitor northern hairy-nosed wombats. The table represents the data shown in Figure 1, so individuals have already been identified and we’re looking at the number of times each was recaptured.</p>
<p> 
 
 </p>
</div>
<div id="simple-capture-recapture-example" class="section level2" number="2.3">
<h2><span class="header-section-number">2.3</span> Simple capture recapture example</h2>
<p>Next we will learn a simple way to make population size estimations. We will use the <code>capwire</code> package to do this. Remember how to install and load a package?</p>
<div class="sourceCode" id="cb48"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb48-1"><a href="molecular-markers.html#cb48-1" aria-hidden="true" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">&quot;capwire&quot;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb49"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb49-1"><a href="molecular-markers.html#cb49-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(capwire)</span></code></pre></div>
<p>The manual for capwire can be found online <a href="https://cran.r-project.org/web/packages/capwire/capwire.pdf">here</a></p>
<p> </p>
<p>Okay, now lets make up some data. Luckily, <em>capwire</em> has a function that allows you to simulate data. Look up the function <code>simCapture</code>. Notice there are three different arguments we need to specify (we can ignore <em>return.cap.probs</em> because it has a default that we do not want to change). Using this function we’ll simulate a population of 300, from which we have 50 samples. This distribution function just means that every individual has an equal probability of being captured.</p>
<div class="sourceCode" id="cb50"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb50-1"><a href="molecular-markers.html#cb50-1" aria-hidden="true" tabindex="-1"></a>sim <span class="ot">&lt;-</span> <span class="fu">simCapture</span>(<span class="at">n=</span><span class="dv">300</span>,<span class="at">s=</span><span class="dv">50</span>,<span class="at">dist.func=</span><span class="fu">drawCapRatesUnif</span>(<span class="fl">0.1</span>,<span class="dv">1</span>)) <span class="co"># Simulate capture data</span></span>
<span id="cb50-2"><a href="molecular-markers.html#cb50-2" aria-hidden="true" tabindex="-1"></a>sim</span></code></pre></div>
<pre><code>##   capture.class No.Ind
## 1             1     44
## 2             2      3</code></pre>
<p>Now that we have simulated data, we can use one of the <code>fitEcm</code> function to estimate population size. What arguments do we need for that function? Notice that the help page tells you what format your data should be in.</p>
<div class="sourceCode" id="cb52"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb52-1"><a href="molecular-markers.html#cb52-1" aria-hidden="true" tabindex="-1"></a>ecm <span class="ot">&lt;-</span> <span class="fu">fitEcm</span>(<span class="at">data=</span>sim,<span class="at">max.pop=</span><span class="dv">500</span>) <span class="co"># estimate population size</span></span></code></pre></div>
<p>The <code>fitEcm</code> page also describes that output. We are most interested in the population size estimate, which is called <code>ml.pop.size</code>. We can extract that using:</p>
<div class="sourceCode" id="cb53"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb53-1"><a href="molecular-markers.html#cb53-1" aria-hidden="true" tabindex="-1"></a>ecm<span class="sc">$</span>ml.pop.size</span></code></pre></div>
<pre><code>## [1] 392</code></pre>
<p>How much uncertainty is there in our estimate? We can use bootstrap resampling to create confidence intervals. Look up the command <code>boostrapCapwire</code></p>
<div class="sourceCode" id="cb55"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb55-1"><a href="molecular-markers.html#cb55-1" aria-hidden="true" tabindex="-1"></a>boot <span class="ot">&lt;-</span> <span class="fu">bootstrapCapwire</span>(<span class="at">x=</span>ecm,<span class="at">bootstraps=</span><span class="dv">1000</span>,<span class="at">CI=</span><span class="fu">c</span>(<span class="fl">0.025</span>,<span class="fl">0.975</span>)) <span class="co"># estimate confidence intervals</span></span>
<span id="cb55-2"><a href="molecular-markers.html#cb55-2" aria-hidden="true" tabindex="-1"></a>boot</span></code></pre></div>
<pre><code>## $ml.pop.size
## [1] 392
## 
## $conf.int
##  2.5% 97.5% 
##   187   500</code></pre>
<p>Using this 95% confidence interval means there is a 95% chance the real answer is within that range. Why is it so large?! What happens if we have more samples?</p>
<p> 
 
 </p>
</div>
<div id="homework-1" class="section level2" number="2.4">
<h2><span class="header-section-number">2.4</span> Homework</h2>
<p>Now that we know how to estimate population sizes, we will use the wombat data from before. Start a new script for this.</p>
<div id="homework-2-write-a-script-that-does-the-following" class="section level3" number="2.4.1">
<h3><span class="header-section-number">2.4.1</span> <em>Homework 2: Write a script that does the following:</em></h3>
<ol style="list-style-type: decimal">
<li><p>Read in “wombats.csv”</p></li>
<li><p>Estimate the total population size.</p></li>
<li><p>Estimate confidence intervals for the total population size.</p></li>
<li><p>Estimate male and female population sizes.</p></li>
<li><p>What is the ratio of males to females in this population?</p></li>
</ol>

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="index.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="genetic-diversity.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["Website.pdf", "Website.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
